docs/csi-controller-attach-detach.md

```diff
@@ -0,0 +1,123 @@
+  # Please refer to the top-level description of nodeSelector for an example
+  nodeSelector: {}
+  # tolerations allow CSI node servers to run on tainted kubernetes nodes
+  # Please refer to the top-level description of tolerations for an example
+  tolerations: []
+  # affinity provides more granular control of which kubernetes nodes will run the CSI node servers
+  # Please refer to the top-level description of affinity for an example
+  affinity: {}
+```
+Please note that where you restrict the node servers to defines what nodes can or cannot mount Pure Storage PVCs. **If
+the node server is not running on a node, you cannot mount Pure Storage volumes on that node.**
+
+### Controller Server
+To restrict where the controller server pods may run, you may set these values:
+```yaml
+controllerServer:
+  # nodeSelector is the simplest way to limit which kubernetes node will run the CSI controller server
+  # Please refer to the top-level description of nodeSelector for an example
+  nodeSelector: {}
+  # tolerations allows the CSI controller servers to run on a tainted kubernetes node
+  # Please refer to the top-level description of tolerations for an example
+  tolerations: []
+  # affinity provides more granular control of which kubernetes node will run the CSI controller server
+  # Please refer to the top-level description of affinity for an example
+  affinity: {}
+```
+The controller pod must be able to access the management endpoint of your Pure Storage backends. Other than that, there
+are no restrictions on where the controller server may run.
+
+### Database
+To restrict where the cockroach-operator, db-deployer, and pso-db pods may run, you may set these values:
+```yaml
+database:
+  # nodeSelector is the simplest way to limit which kubernetes nodes will run the database-related pods
+  # Please refer to the top-level description of nodeSelector for an example
+  nodeSelector: {}
+  # tolerations allows the database-related pods to run on tainted kubernetes nodes
+  # Please refer to the top-level description of tolerations for an example
+  tolerations: []
+  # affinity provides more granular control of which kubernetes nodes will run the database-related pods
+  # Please refer to the top-level description of affinity for an example
+  affinity: {}
+```
+The cockroach-operator, db-deployer, and pso-db pods must all be able to access the management endpoint of your Pure
+Storage backends. These nodes must also be time-synced to less than a 500 ms interval. Please also ensure that these
+nodes are able to sustain a small distributed database workload.
+
+## Examples
+### Restrict all REST-requiring pods to master nodes only
+```yaml
+# Allow node servers to run on all nodes (so that Pure PVCs can be mounted anywhere)
+nodeServer:
+  nodeSelector: {}
+  tolerations: []
+  affinity: {}
+
+controllerServer:
+  # Require that the controller server runs on control plane nodes
```

Comment:
[Suggestion]
Please correct me if I'm wrong. I *THINK* there is no universal label to identify a K8s master node. Different cluster orchestrator may use different labels.

Is it possible we add a comment to list two common labels, maybe the label used by `kubeadm`, and one by Rancher?

What do you think?

Reply:
I agree, I'll check what the common ones are and add them in.

Reply:
@carysyd so I did some more searching and thinking: I don't think kubeadm actually sets any labels for master nodes, it only adds taints on the master nodes. I will make sure to add a note saying that your labels may vary.